cmd/manager/server.go

```diff
@@ -95,9 +91,9 @@ func run() {
 		log.Fatalf("Failed to get watch namespace: %v", err)
 	}
 
-	leaderElectionNS, err := getLeaderElectionNamespace()
-	if err != nil {
-		log.Fatalf("Failed to get leader election namespace: %v", err)
+	leaderElectionNS, leaderElectionEnabled := getLeaderElectionNamespace()
+	if !leaderElectionEnabled {
```

Comment:
Also for your original concern, would it solve the issue if you attach the debug process after it acquires the lock

Reply:
It wouldn't. The problem is that the leader election thread needs to run constantly so it can keep updating the lock config map. If it doesn't do that before the lock expires, it's no longer the leader and must terminate immediately. And that's exactly what happens when you allow the thread to run after being suspended for too long.